lib/services/dashd.js

```diff
@@ -776,46 +776,57 @@ Dash.prototype._checkSyncedAndSubscribeZmqEvents = function(node) {
 
 Dash.prototype._subscribeZmqEvents = function(node) {
   var self = this;
-  node.zmqSubSocket.subscribe('hashblock');
-  node.zmqSubSocket.subscribe('rawtx');
-  node.zmqSubSocket.subscribe('rawtxlock');
-  node.zmqSubSocket.on('message', function(topic, message) {
-    var topicString = topic.toString('utf8');
-    if (topicString === 'rawtxlock') {
-      self._zmqTransactionLockHandler(node, message);
-    } else if (topicString === 'rawtx') {
-      self._zmqTransactionHandler(node, message);
-    } else if (topicString === 'hashblock') {
-      self._zmqBlockHandler(node, message);
-    }
+  node.zmqSubSocket.subscribe(DashdZMQ.TOPICS.hashblock);
+  node.zmqSubSocket.on(DashdZMQ.TOPICS.hashblock, (message) => {
+    self._zmqBlockHandler(node, Buffer.from(message, 'hex'));
+  });
+
+  node.zmqSubSocket.subscribe(DashdZMQ.TOPICS.rawtx);
+  node.zmqSubSocket.on(DashdZMQ.TOPICS.rawtx, (message) => {
+    self._zmqTransactionHandler(node, Buffer.from(message, 'hex'));
+  });
+
+  node.zmqSubSocket.subscribe(DashdZMQ.TOPICS.rawtxlock);
+  node.zmqSubSocket.on(DashdZMQ.TOPICS.rawtxlock, (message) => {
+    self._zmqTransactionLockHandler(node, Buffer.from(message, 'hex'));
   });
 };
 
 Dash.prototype._initZmqSubSocket = function(node, zmqUrl) {
   var self = this;
-  node.zmqSubSocket = zmq.socket('sub');
+  return new Promise((resolve) => {
+    const [protocol, path] = zmqUrl.split('//');
```

Comment:
Let's make use of your new url functionality
https://github.com/dashevo/js-dashd-zmq/pull/1